### PR TITLE
feat: Filter actions for notes through Date Priority and Labels

### PR DESCRIPTION
**Pull request by dev-enforced**
In this PR one can find the following features added to Notes App:
- Filter Modal Screen that can be accessed through the click of a button
- Filter by Priority
- Filter By Date
- Filter By Labels
- Documentation for UNOTE

### CODE REVIEW

**Review by ankit9015 — APPROVED** [review]
Code is well structured and looks good to merge

File: src/components/FilterModal/FilterModal.jsx [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import React from "react";
+import { useFilters, useNotes } from "context";
+import "./FilterModal.css";
+const FilterModal = ({ checkDisplayModal }) => {
+  const { noteLabels } = useNotes();
+  const { currentFilterState, filterStateUpdate } = useFilters();
+  const {
+    sortBy: { oldToRecent, recentToOld },
```
Comment (hinanshisuthar) [inline src/components/FilterModal/FilterModal.jsx:8]: Good way to destructure here itself

File: src/components/FilterModal/FilterModal.jsx [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+                    <span className="fs-1-5">Date: Old To Recent</span>
+                  </label>
+                </li>
+              </ul>
+            </div>
+            <div className="filter-container">
+              <h5>SORT BY LABELS</h5>
+              <ul className="gentle-flex-gap g-flex-wrap">
+                {noteLabels &&
+                  noteLabels?.map((everyLabel) => {
```
Comment (hinanshisuthar) [inline src/components/FilterModal/FilterModal.jsx:81]: Good use of optional chaining

File: src/constants/dateFormatter.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+const dateFormatter = new Intl.DateTimeFormat("en-us", {
```
Comment (hinanshisuthar) [inline src/constants/dateFormatter.js:1]: Learnt a new way of generating date

File: src/pages/NotesHome/NotesHome.jsx [hunk 4]
```diff
@@ -1,16 +1,32 @@
-import React from "react";
-import { SideNavigation, NoteInput, NoteDisplay,PinnedNotesSection } from "components";
+import React, { useState } from "react";
+import {
+  SideNavigation,
+  NoteInput,
+  NoteDisplay,
+  PinnedNotesSection,
+  FilterModal
+} from "components";
```
Comment (hinanshisuthar) [inline src/pages/NotesHome/NotesHome.jsx:8]: Liked the use of barrel imports

**Review by hinanshisuthar — APPROVED** [review]
Gone though the code
- clean code written
- good practices followed overall
- Functionalities work as intended
- Good logic implementation

**Pull request merged**